File: nom-api/Nom.Orch/Models/MealPlan/MealPlanCreateResponseModel.cs
// File: Nom.Orch/Models/MealPlan/MealPlanCreateResponseModel.cs

using System;

namespace Nom.Orch.Models.MealPlan
{
    public class MealPlanCreateResponseModel
    {
        public long Id { get; set; }
        public long HouseholdId { get; set; }
        public long AuthorId { get; set; }
        public DateOnly Date { get; set; }
        public long MealTypeId { get; set; }
        public string? Title { get; set; }
        public string? Notes { get; set; }
        public long? RecipeId { get; set; }
        public DateTime CreatedDate { get; set; }
    }
} 

File: nom-ui/src/app/meal-plan/models/meal-plan-create-response.model.ts
// File: nom-ui/src/app/meal-plan/models/meal-plan-create-response.class.ts

import { IMealPlanCreateResponseModel } from './meal-plan-create-response.model.interface';

export class MealPlanCreateResponseModel implements IMealPlanCreateResponseModel {
    id = 0;
    householdId = 0;
    date: Date = new Date();
    mealTypeId = 0;
    title = '';
    notes?: string;
    recipeId?: number;
    createdDate: Date = new Date();

    constructor(data?: Partial<IMealPlanCreateResponseModel>) {
        if (data) {
            Object.assign(this, data);
        }
    }
} 